### ISSUE TITLE
bug: Ion-refresher remains disabled even when the disabled value bound is true

### ISSUE THREAD

**Issue by andrasz** [ionic-team/ionic#20339]
# Bug Report

**Ionic version:**
**4.x**

**Current behavior:**
When changing the value of the disabled attribute from false to true, the ion-refresher won't be enabled. The issue comes up, when the disabled state is changed from a scroll event.

**Expected behavior:**
When the disabled property is true, the ion-refresher should be enabled.

**Steps to reproduce:**
In the sample code, initially the ion-refresher works. By pulling down the list, it triggers the ionRefresh event. If you scroll down the list the refresher should be disabled until you scroll up to the top of the list where the disabled property will be false again. At this point the refresher wont trigger the ionRefresh event when pulled down again even though the disabled attribute is false.

**Related code:**
*Template*
```html
  <ion-refresher [disabled]="dis" (ionRefresh)="refreshAppointments($event)" slot="fixed">
    <ion-refresher-content pullingIcon=null refreshingSpinner=null>
    </ion-refresher-content>
  </ion-refresher>
  <ul class="list" style="max-height: 150px;overflow: auto;">
    <li>Some list 1</li>
    <li>Some list 2</li>
    <li>Some list 3</li>
  </ul>
```
*Typescript*
```javascript
dis = false;
refreshAppointments(event) {
  setTimeout(() => {
    event.target.complete();
  }, 3000);
}
ionViewDidEnter() {
  const el = document.getElementsByClassName('list')['0']; // query the list
  if (el) {
    el.addEventListener('scroll', e => {
      this.dis = true;
      if (e.target.scrollTop === 0) { // enable ion-refresher only when the list is scrolled to the top
        this.dis = false;
      }
    });
  }
}
```

**Other information:**
Another issue I found about this topic was closed without any solution here:
https://github.com/ionic-team/ionic/issues/17965

**Ionic info:**

```
Ionic:

   Ionic CLI                     : 5.4.9
   Ionic Framework               : @ionic/angular 4.11.10
   @angular-devkit/build-angular : 0.801.3
   @angular-devkit/schematics    : 8.1.3
   @angular/cli                  : 8.1.3
   @ionic/angular-toolkit        : 2.1.2

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v10.15.3
   npm    : 6.11.3
   OS     : Windows 10
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you provide a repo with the code required to reproduce this issue? I am unable to reproduce the issue using the code provided.

Here are the steps I am following:

1. Pull down on screen, pull to refresh should work (expected).
2. Scroll down on list.
3. Pull down on screen, pull to refresh should _not_ work (expected).
4. Scroll all the way back to the top.
5. Pull down on screen, pull to refresh should still not work (unexpected). In this case, I am seeing pull to refresh work properly.

**Comment by andrasz** [comment]
Hi @liamdebeasi, Thanks for looking into the issue! As I tried to create the repo it was working at first, so I found out that this only happens on a mobile environment. Maybe it has something to do with touch events, I'm not sure.
Here's the repo I put together: https://github.com/andrasz/ion-refresher-bug-reproduce.git
The steps you followed were right, when using a mobile device the issue should come up!
